igneous/task_creation.py

```diff
@@ -623,6 +624,62 @@ def create_hypersquare_consensus_tasks(task_queue, src_path, dest_path, volume_m
     task_queue.insert(task)
   task_queue.wait()
 
+def create_mask_affinity_map_tasks(task_queue, aff_input_layer_path, aff_output_layer_path, 
+        aff_mip, mask_layer_path, mask_mip, output_block_start, output_block_size, grid_size ):
+    """
+    affinity map masking block by block. The block coordinates should be aligned with 
+    cloud storage. 
+    """
+    for z in tqdm(range(grid_size[0]), desc='z loop'):
```

Comment:
Could `grid_size` be computed from a CloudVolume instance? It might make using this simpler.

Reply:
Check out other functions that use `xyzrange` that might be the single loop functionality you're looking for.

Reply:
In all of the datasets I processed, the volume range is larger than what we need to process. So computing `grid_size` might not be useful at all!

Reply:
If you'd like to run on restricted bounding boxes, check out the recent changes I made to the downsample function. I added a `bounds` flag.

Reply:
tks, I changed to `bounds`

Reply:
changed back to offset and shape since bounds could not be serialized by JSON.

Reply:
That's fine. Typically i like to pass in bbox and then do something like:

```
bbox.size3() # shape
bbox.minpt.clone() # offset
```

Matter of style though.